src/mgmt/srv_drpc.c

```diff
@@ -1973,8 +1973,12 @@ ds_mgmt_smd_free_dev(Ctl__SmdDevice *dev)
 		D_FREE(dev->ctrlr->fw_rev);
 		D_FREE(dev->ctrlr->vendor_id);
 		D_FREE(dev->ctrlr->pci_dev_type);
-		if (dev->ctrlr->namespaces != NULL)
+		if (dev->ctrlr->namespaces != NULL) {
 			D_FREE(dev->ctrlr->namespaces[0]);
+			D_FREE(dev->ctrlr->namespaces);
+			dev->ctrlr->namespaces   = NULL;
```

Comment:
[style] D_FREE() inside will assign NULL after free.

Reply:
Thank you I was following the pattern in list_devs but I will remove both NULL assignments when I visit this area next.